### PR TITLE
codeintel: Move graph export task to uploads service

### PR DESCRIPTION
**Pull request by efritz**
This PR moves a background task related to ranking/exports in the codenav service to teh uploads service. This is not a feature that we've exposed to end users yet. https://github.com/sourcegraph/deploy-sourcegraph-cloud/pull/17465 must land prior to this PR.

## Test plan

Tested locally.

### CODE REVIEW

File: enterprise/internal/codeintel/uploads/observability.go [hunk 1]
```diff
@@ -77,6 +82,33 @@ func newOperations(observationContext *observation.Context) *operations {
...
+			Name: name,
+			Help: help,
+		})
+
+		observationContext.Registerer.MustRegister(counter)
+		return counter
+	}
+
+	numUploadsRead := counter(
+		"src_codeintel_codenav_ranking_uploads_read_total",
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/observability.go:96]: Update these (and in monitoring)

File: enterprise/internal/codeintel/uploads/config.go [hunk 2]
```diff
@@ -102,3 +102,18 @@ func (c *expirationConfig) Load() {
...
+	env.BaseConfig
+
+	RankingInterval    time.Duration
+	NumRankingRoutines int
+}
+
+var ConfigExportInst = &exportConfig{}
+
+func (c *exportConfig) Load() {
+	// TODO - rename once advertised / renamed on dotcom
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/config.go:116]: Note!

File: enterprise/internal/codeintel/uploads/init.go [hunk 3]
```diff
@@ -44,18 +51,46 @@ type serviceDependencies struct {
 	gsc         GitserverClient
 }
 
+var (
+	bucketName                   = env.Get("CODEINTEL_CODENAV_RANKING_BUCKET", "lsif-pagerank-experiments", "The GCS bucket.")
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/init.go:55]: Rename these to fit new service

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ef/move-export/compare-report.html;tab=live_object) for a full comparison between the commits a62e39f and d1c014a or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: enterprise/internal/codeintel/uploads/init.go [hunk 4]
```diff
@@ -44,18 +51,46 @@ type serviceDependencies struct {
...
+		return client.Bucket(bucketName)
+	}()
+
 	svc := newService(
 		store,
 		repoStore,
 		lsifStore,
 		deps.gsc,
+		rankingBucket,
 		nil, // written in circular fashion
```
Comment (Strum355) [inline enterprise/internal/codeintel/uploads/init.go:94]: Hope we find a solution for this :grimacing:

**Review by Strum355 — APPROVED** [review]

File: enterprise/internal/codeintel/uploads/init.go (hunk 4 above)
Comment (efritz) [reply enterprise/internal/codeintel/uploads/init.go:94]: The line that wasn't changed by this PR?

Comment (Strum355) [reply enterprise/internal/codeintel/uploads/init.go:94]: Oh I know, a drive by comment

Comment (efritz) [reply enterprise/internal/codeintel/uploads/init.go:94]: Just making sure it wasn't the bucket line 😅

**Review by Numbers88s — APPROVED** [review]

**Pull request merged**